### PR TITLE
feat(TNO-1401 - Openshift kustomize code optimization)

### PR DESCRIPTION
**Pull request by kulpree**
Openshift (kustomize) code optimization for the tno resource deployment.

### CODE REVIEW

**Review by Fosol — COMMENTED** [review]
Thanks @kulpree for cleaning up the file structure.  I'll review it tonight/tomorrow morning.  Quick glance so far seems great.

File: db/kafka/scripts/topic-add.sh [hunk 1]
```diff
@@ -34,7 +34,7 @@ fi
 eval set -- "$PARSED"
 
 # Default values.
-environment=local bootstrap=${BOOTSTRAP:-tno-kafka-broker:29094} partitions=3 replication=3 index=0
+environment=local bootstrap=${BOOTSTRAP:-kafka-broker-0:29094} partitions=3 replication=3 index=0
```
Comment (Fosol) [inline db/kafka/scripts/topic-add.sh:37]: We'll need to update with overrides for our local environments.  The original script helped developers get their environment setup.

File: db/kafka/scripts/topic-add.sh [hunk 2]
```diff
@@ -105,5 +105,5 @@ export index
 if [ "$environment" = "local" ]; then
   docker exec -i tno-broker bash -c "/bin/kafka-topics --create --topic $topic --bootstrap-server $bootstrap --partitions $partitions --replication-factor $replication"
 else
-  oc rsh -n 9b301c-$environment tno-kafka-broker-$index bash -c "/bin/kafka-topics --create --topic $topic --bootstrap-server $bootstrap --partitions $partitions --replication-factor $replication"
+  oc rsh -n 9b301c-$environment kafka-broker-$index bash -c "/bin/kafka-topics --create --topic $topic --bootstrap-server $bootstrap --partitions $partitions --replication-factor $replication"
```
Comment (Fosol) [inline db/kafka/scripts/topic-add.sh:108]: Same issue here

File: openshift/kustomize/api/overlays/prod/kustomization.yaml [hunk 3]
```diff
@@ -133,7 +133,7 @@ patches:
         value: 1Gi
       - op: replace
         path: /spec/template/spec/containers/0/image
-        value: image-registry.openshift-image-registry.svc:5000/9b301c-tools/api:prod
+        value: image-registry.openshift-image-registry.svc:5000/9b301c-prod/api:prod
```
Comment (Fosol) [inline openshift/kustomize/api/overlays/prod/kustomization.yaml:136]: We don't push images to each environment, we keep them all in the tools namespace.

File: openshift/kustomize/kafka/zookeeper/overlays/prod/kustomization.yaml [hunk 4]
```diff
@@ -16,7 +16,7 @@ patches:
         value: 3
       - op: replace
         path: /spec/template/spec/containers/0/image
-        value: image-registry.openshift-image-registry.svc:5000/9b301c-tools/zookeeper:prod
+        value: image-registry.openshift-image-registry.svc:5000/9b301c-prod/zookeeper:prod
```
Comment (Fosol) [inline openshift/kustomize/kafka/zookeeper/overlays/prod/kustomization.yaml:19]: Images are kept in the tools namespace

File: openshift/scripts/create-environment.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
+# Create policy for prod to pull images
+oc policy add-role-to-user system:image-puller system:serviceaccount:9b301c-prod:default -n 9b301c-tools
```
Comment (Fosol) [inline openshift/scripts/create-environment.sh:2]: Thanks for putting this together.

**Review by Fosol — CHANGES_REQUESTED** [review]

File: db/kafka/scripts/topic-add.sh (hunk 1 above)
Comment (kulpree) [reply db/kafka/scripts/topic-add.sh:37]: Thanks! I will add new script tied with production deployment script.

**Comment by Fosol** [comment]
@jdtoombs  Is this ready now?

File: openshift/kustomize/api/overlays/prod/kustomization.yaml (hunk 3 above)
Comment (kulpree) [reply openshift/kustomize/api/overlays/prod/kustomization.yaml]: Image pull updated back to tools.

File: openshift/kustomize/kafka/zookeeper/overlays/prod/kustomization.yaml (hunk 4 above)
Comment (kulpree) [reply openshift/kustomize/kafka/zookeeper/overlays/prod/kustomization.yaml]: Updated back to tools namespace.

File: db/kafka/scripts/topic-add.sh (hunk 2 above)
Comment (kulpree) [reply db/kafka/scripts/topic-add.sh]: Pushed both changes and added a new file for Openshift deployments.

**Comment by kulpree** [comment]
All the comments are resolved. Please validate and merge.

**Review by Fosol — APPROVED** [review]

**Pull request merged**